src/service/service.py

```diff
@@ -121,7 +121,7 @@ async def message_generator(user_input: StreamInput) -> AsyncGenerator[str, None
                 # Empty content in the context of OpenAI or Anthropic usually means
```

Comment:
When retesting the fix, I noticed another issue, which I will have to look at more closely. As opposed to OpenAI models, when Claude uses tool calls, `content` is not empty, but has a value like this: `[{'id': 'toolu_01H6DqC7db6kEDMthUa7LrPx', 'input': {}, 'name': 'WebSearch', 'type': 'tool_use', 'index': 0}]` . The empty content check doesn't seem sufficient.

Reply:
Oh interesting, I hadn’t noticed that. Do you think we need to fix here before merging? Have a fix in mind?

Reply:
I was thinking which approach will be best, if to keep only text content or to only filter tool use content and keep the rest. I settled on the latter. When ChatMessage abstraction is removed and `convert_message_content_to_string()` goes with it, UI will have to adapt to the possibility of streamed content not being just strings, but also dictionary content items (possibly even images).